scrapy/spiders/crawl.py

```diff
@@ -105,15 +115,30 @@ def _requests_to_follow(self, response):
 
     def _callback(self, response, **cb_kwargs):
         rule = self._rules[response.meta["rule"]]
-        return self._parse_response(
+        return self.parse_with_rules(
             response, rule.callback, {**rule.cb_kwargs, **cb_kwargs}, rule.follow
         )
 
     def _errback(self, failure):
         rule = self._rules[failure.request.meta["rule"]]
         return self._handle_failure(failure, rule.errback)
 
-    async def _parse_response(self, response, callback, cb_kwargs, follow=True):
+    def parse_with_rules(self, response, callback, cb_kwargs, follow=True):
+        direct_call = True
+        return self._parse_response(response, callback, cb_kwargs, follow, direct_call)
```

Comment:
This will break in cases where `_parse_response` is overridden, because the overridden method will not expect an additional parameter (`direct_call`).

Reply:
You are right. I did not think about that case. I fixed it in the last commit passing `direct_call` in `cb_kwargs ` without changing the original method signature.